client/app/components/dynamic-form/DynamicForm.jsx

```diff
@@ -56,10 +56,11 @@ class DynamicForm extends React.Component {
   constructor(props) {
     super(props);
 
+    const filledExtraFields = filter(props.fields, field => field.extra && field.initialValue !== undefined);
```

Comment:
Perhaps use `some`?

Reply:
Notice that if a user sets an extra field but then clear the field - it won't be undefined but '' hence the section would open needlessly.

Maybe check `!isEmpty(field.initialValue)` instead?

Reply:
It is a bit trickier to determine whether a field is empty :confused:, `undefined` was the easiest I thought about (but there is the issue you mentioned). `isEmpty`won't work neither for number nor for booleans. I'm thinking about an alternative option as I would like to keep it as a DynamicForm behavior :grimacing: